src/python_testing/TC_CADMIN_1_3_4.py

```diff
@@ -0,0 +1,376 @@
+        await self.write_nl_attr(th=self.th1, attr_val=nl_attribute)
+        await self.read_nl_attr(th=self.th1, attr_val=nl_attribute)
+
+        self.step(4)
+        # Establishing TH2
+        th2_certificate_authority = self.certificate_authority_manager.NewCertificateAuthority()
+        th2_fabric_admin = th2_certificate_authority.NewFabricAdmin(vendorId=0xFFF1, fabricId=self.th1.fabricId + 1)
+        self.th2 = th2_fabric_admin.NewController(nodeId=2, useTestCommissioner=True)
+        await self.th2.CommissionOnNetwork(
+            nodeId=self.dut_node_id, setupPinCode=params.setupPinCode,
+            filterType=ChipDeviceCtrl.DiscoveryFilterType.LONG_DISCRIMINATOR, filter=1234)
+        rcac = self.th2.get_rcac()
+
+        th2_rcac_decoded = TLVReader(rcac).get()["Any"][9]
+
+        self.step(5)
+        # TH_CR1 reads the Fabrics attribute from the Node Operational Credentials cluster using a fabric-filtered read
+        th1_fabric_info = await self.get_fabrics(th=self.th1)
+
+        # Verify that the RootPublicKey matches the root public key for TH_CR1 and the NodeID matches the node ID used when TH_CR1 commissioned the device.
+        await self.send_single_cmd(dev_ctrl=self.th1, node_id=self.dut_node_id, cmd=Clusters.GeneralCommissioning.Commands.ArmFailSafe(10))
+        th1_cam_rcac = TLVReader(base64.b64decode(
+            self.certificate_authority_manager.activeCaList[0]._persistentStorage._jsonData["sdk-config"]["f/1/r"])).get()["Any"][9]
+        if th1_fabric_info[0].rootPublicKey != th1_cam_rcac:
+            asserts.fail("public keys from fabric and certs for TH1 are not the same")
+        if th1_fabric_info[0].nodeID != self.dut_node_id:
+            asserts.fail("DUT node ID from fabric does not equal DUT node ID for TH1 during commissioning")
+
+        # Expiring the failsafe timer in an attempt to clean up before TH2 attempt.
+        await self.th1.SendCommand(self.dut_node_id, 0, Clusters.GeneralCommissioning.Commands.ArmFailSafe(0))
+
+        self.step(6)
+        # TH_CR2 reads the Fabrics attribute from the Node Operational Credentials cluster using a fabric-filtered read
+        th2_fabric_info = await self.get_fabrics(th=self.th2)
+
+        # Verify that the RootPublicKey matches the root public key for TH_CR2 and the NodeID matches the node ID used when TH_CR2 commissioned the device.
+        await self.send_single_cmd(dev_ctrl=self.th2, node_id=self.dut_node_id, cmd=Clusters.GeneralCommissioning.Commands.ArmFailSafe(10))
+
+        if th2_fabric_info[0].rootPublicKey != th2_rcac_decoded:
+            asserts.fail("public keys from fabric and certs for TH1 are not the same")
+        if th2_fabric_info[0].nodeID != self.dut_node_id:
+            asserts.fail("DUT node ID from fabric does not equal DUT node ID for TH1 during commissioning")
+
+        await self.th2.SendCommand(self.dut_node_id, 0, Clusters.GeneralCommissioning.Commands.ArmFailSafe(0))
```

Comment:
same - I think both of these can be removed.

Reply:
and below.

Reply:
Thank you, these have been removed as well now I believe